raiden/transfer/channel.py

```diff
@@ -219,19 +222,26 @@ def is_channel_usable_for_new_transfer(
     - The channel has capacity.
     - The number of locks can be claimed on-chain.
     - The transfer amount does not overflow.
+    - lock_timeout, if provided, is within allowed range (reveal_timeout, settle_timeout]
 
     The number of locks has to be checked because the gas usage will increase
     linearly with the number of locks in it, this has to be limited to a value
     lower than the block gas limit constraints.
     """
     pending_transfers = get_number_of_pending_transfers(channel_state.our_state)
     distributable = get_distributable(channel_state.our_state, channel_state.partner_state)
+    lock_timeout_valid = True
+    if lock_timeout and (
+        lock_timeout > channel_state.settle_timeout or lock_timeout <= channel_state.reveal_timeout
+    ):
+        lock_timeout_valid = False
```

Comment:
These 4 lines can be replaced with
```
lock_timeout_valid = (
    lock_timeout <= channel_state.settle_timeout and
    lock_timeout > channel_state.reveal_timeout
)
```

Also, my suggestion changes the comparison operators to match the conditions you've removed from the caller function

Reply:
You version gives this:
```
(raiden) admins-MacBook-Pro-2:raiden admin$ make lint
mypy raiden
raiden/transfer/channel.py:239: error: Unsupported operand types for >= ("BlockTimeout" and "None")
raiden/transfer/channel.py:239: note: Left operand is of type "Optional[BlockTimeout]"
raiden/transfer/channel.py:240: error: Unsupported operand types for < ("BlockTimeout" and "None")
raiden/transfer/channel.py:240: note: Left operand is of type "Optional[BlockTimeout]"
make: *** [mypy] Error 1
```

changing it to:
```
    lock_timeout_valid = lock_timeout is None or (
        lock_timeout <= channel_state.settle_timeout and
        lock_timeout > channel_state.reveal_timeout
    )
```
solved the lint problem but my IDE (PyCharm) comlains "Simplifu chained comparison"

I changed to the second version but it looks buggy.